src/passes/Precompute.cpp

```diff
@@ -161,6 +161,8 @@ struct Precompute : public WalkerPass<PostWalker<Precompute, UnifiedExpressionVi
   void visitExpression(Expression* curr) {
     // TODO: if get_local, only replace with a constant if we don't care about size...?
     if (curr->is<Const>() || curr->is<Nop>()) return;
+    // Until engines implement v128.const, do not try to precompute v128 expressions
```

Comment:
Please add to the comment about the size issue - it does seem like this is a new factor we need to consider here, where precomputing may actually increase the size and be detrimental.